web/src/components/AppBar/ApiKeyItem.vue

```diff
@@ -12,35 +12,68 @@
       <v-text-field
         ref="apiKey"
         v-model="apiKey"
-        label="Api Key"
-        :readonly="true"
+        label="API Key"
+        readonly
         append-outer-icon="mdi-content-copy"
-        @click:append-outer="copyApiKey"
+        @click:append-outer="copyToClipboard"
       />
     </v-list-item-content>
   </v-list-item>
 </template>
 
 <script>
-import { mapActions, mapState } from 'vuex';
+import girderRest from '@/rest';
 
 export default {
   name: 'ApiKeyItem',
-  computed: {
-    ...mapState('girder', ['apiKey']),
+  data() {
+    return {
+      apiKey: null,
+    };
   },
   created() {
-    this.fetchApiKey();
+    this.fetch();
   },
   methods: {
-    copyApiKey() {
-      const { input: value } = this.$refs.apiKey.$refs;
-      value.focus();
+    copyToClipboard() {
+      const vTextFieldComponent = this.$refs.apiKey;
+      // v-text-field provides some internal refs that we can use
+      // one is "input", which is the actual <input> DOM element that it uses
+      const inputElement = vTextFieldComponent.$refs.input;
+      inputElement.focus();
       document.execCommand('selectAll');
-      value.select();
+      inputElement.select();
       document.execCommand('copy');
     },
-    ...mapActions('girder', ['fetchApiKey', 'reloadApiKey']),
+    async fetch() {
+      const { data } = await girderRest.get(
+        'api_key', {
+          params: {
+            // eslint-disable-next-line import/no-named-as-default-member
+            userId: girderRest.user._id,
+            limit: 50,
+            sort: 'name',
+            sortdir: 1,
+          },
+        },
+      );
+      let [dandiKey] = data.filter((key) => key.name === 'dandicli');
+      if (!dandiKey) {
+        // create a key
+        const { data2 } = await girderRest.post('api_key', null, {
```

Comment:
Does this work? Since you're using destructuring assignment, this expects a property of `girderRest` to be called `data2` (i.e. this is the same as `const data2 = await girderRest.post(...).data2`).

You could do `const { data: data2 }` to avoid a name conflict, but `data2` is a very magic and non-descriptive variable name. If I look at that code in the future, it'd be hard to understand why it's "2" and what information that "2" is trying to deliver to the reader. What if we add more Ajax calls in the future? Will those be `data3`, `data4`, etc.? :smiley: 

Since you don't need the original `data` (from the GET request) down at this point in the code, you could just originally declare `let data`, which would allow you to reassign to the same name here.

Reply:
ok. Got it. Thanks. I was doing a quick fix when cli test failed and yelled about the naming of "data"